base/services/audiosrv/services.c

```diff
@@ -1,13 +1,15 @@
 /*
- * PROJECT:          ReactOS
- * LICENSE:          GPL - See COPYING in the top level directory
- * FILE:             base/services/audiosrv/services.c
- * PURPOSE:          Audio Service Plug and Play
- * COPYRIGHT:        Copyright 2009 Johannes Anderwald
+ * PROJECT:     ReactOS
+ * LICENSE:     GPL - See COPYING in the top level directory
```

Comment:
```suggestion
 * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
```

Reply:
I will have to do it in all files. Therefore I will not accept the suggestion via webinterface, but I add it myself.

Reply:
But is "GPL - See COPYING in the top level directory" really GPL2+? I will add @ColinFinck for certainty here.

Reply:
That file has an exception at the footer of it, so moving it to play GPL will make the code less usable.

Reply:
@learn-more , you wanted to help resolving this discussion. Unfortunately I do not understand entirely yet in which direction you want to push. You want me to **not** change it, right?

Reply:
I am not a lawyer, but I am pretty sure the last paragraph in that file is not plain gpl, so you probably want something slightly less evil like the LGPL.

It might be a good idea to see with which other occurrences of this were replaced.

Reply:
That is still not the clear answer I hoped for. I will wait for @ColinFinck s reply before doing any action on this.
If the doubt would remain after Colins reply, then my personal preference would be to simply not change it.

Reply:
@JoachimHenze The primary problem here is that "GPL - See COPYING in the top level directory" doesn't equal "GPL-2.0-**or-later**".
The license text is just about GPLv2 and several developers have expressed their objections against GPLv3 in the past, explicitly making their code GPL-2.0-only. Unless you ask the developers of this source, you may only change this line to "GPL-2.0-only".

@learn-more You are right that our "COPYING" file in the top-level directory indeed has additional lines compared to a regular GPLv2. I have laid out in https://www.youtube.com/watch?v=OTVVz9VLisU why I think that these lines are superfluous _for ReactOS_ and don't grant any additional rights over the regular GPLv2. I even went as far as saying that the effects of LGPL and GPL are identical _when implementing a Windows-compatible component for ReactOS_.

That said, I'm not a lawyer. You can again only be sure about the intended license when asking the developers of this particular source. Chances are high that they didn't even know about the additional exception in our "COPYING" file when writing this code. They may also agree to relicense it under LGPL or even more permissive if you ask them.

Note that despite my claim that the effects of LGPL and GPL are identical _for us_, you can't just take a GPL-licensed source and slap an LGPL license on it. The source may later be used in a different context where the effects of LGPL and GPL are different.
You could however always restrict the license even more, so replacing "GPL - See COPYING in the top level directory" by "GPL-2.0-only" is definitely possible (effectively removing the additional exceptions from our "COPYING" file).


**TL;DR:** If in doubt, leave this unchanged. If you really need an SPDX line, choose "GPL-2.0-only". If you need more than that, ask the original developers.

Reply:
Ok, I will leave it like it is then.